src/planscape/forsys/views.py

```diff
@@ -271,6 +290,73 @@ def generate_project_areas_for_a_single_scenario(
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
 
+@csrf_exempt
+def generate_project_areas_prototype(
+        request: HttpRequest) -> HttpResponse:
+    try:
+        body = json.loads(request.body)
+        if not 'stand' in body or not 'project' in body:
+            raise ValueError(
+                "Either the forsys stand or project dictionary is missing")
+        forsys_output_json = {
+            'stand': body['stand'],
+            'project': body['project']
+        }
+
+        user_id = body.get('user_id', None)
+        if not (isinstance(user_id, int)):
+            raise ValueError("Must specify user_id as an integer")
+
+        scenario_id = body.get('scenario_id', None)
+        if not (isinstance(scenario_id, int)):
+            raise ValueError("Must specify scenario_id as an integer")
+
+        weighted_priorities = {
+            'storage': 5.0,
+            'california_spotted_owl': 2.0,
+            'functional_fire': 1.0,
+            'forest_structure': 2.0,
+            'max_sdi': 1.0
+        }
+
+        # TODO: move this to a HTTP call in the lambda as part of scenario creation
+        scenario = Scenario.objects.get(id=scenario_id)
```

Comment:
Does the Scenario's owner need to be validated against the user?

And if not, is it necessary to return the user?

Reply:
I see!
Another option: could get_generation_request_params be instantiated with scenario.owner_pk?

Reply:
yes, i was thinking about that as well. updated